### PR TITLE
Relocate files in the API Reference to Explanation/Tutorials

### PR DESCRIPTION
**Pull request by eh111eh**
#### References to other Issues or PRs

Issue #25607

#### Brief description of what is fixed or changed
Moved files in the API Reference to the appropriate location (Explanation, Tutorials) as discussed in Issue #25607.

Also referred to https://diataxis.fr.

#### Other comments
@tjstienstra @moorepants @ricdigi 

#### Release Notes

* physics.vector
  * Moved files in the API Reference to Explanation
* physics.biomechanics
  * Moved files in the API Reference to Explanation
* physics.mechanics
  * Moved files in the API Reference to Explanation and Tutorials

### CODE REVIEW

**Comment by moorepants** [comment]
I wonder if the side bar should have this organization:

- Explanation
  - Modules
    - Physics
      - Vector
      - Mechanics
      - Biomechanics

**Comment by moorepants** [comment]
You'll need to include URL redirect for all of these, so that we don't break the long standing URLs in search and other websites.

**Comment by eh111eh** [comment]
> I think we can improve this listing on the side: image
> 
> It should probably show "Physics" in the list and if you click the dropdown arrow it would show the subsections: Vector, Mechanics, Biomechanics, etc.
> 
> Also, the list should be ordered in some organized way, glossary should probably be last, for example.

I rearranged in the order:
- Explanations
  - Best Practices
  - Gotchas and Pitfalls
  - Solve Output by Type
  - Physics
     - The Physics Vector
     - Classical Mechanics
     - Introduction to Biomechanics Modeling
  - SymPy Special Topics
  - List of active deprecations
  - Glossary

File: doc/src/conf.py [hunk 1]
```diff
@@ -81,6 +81,23 @@
     "tutorial/matrices": "../tutorials/intro-tutorial/matrices.html",
     "tutorial/manipulation": "../tutorials/intro-tutorial/manipulation.html",
 
+    "modules/physics/vector/index": "../explanation/Modules/Physics/vectors/index.html",
```
Comment (moorepants) [inline doc/src/conf.py:84]: I think we should keep all directory names lower case, as that is standard in most (maybe all) of SymPy's code.

**Comment by moorepants** [comment]
This looks good. I have the directory name commetn above and then looking at:

- Explanations
  - Best Practices
  - Gotchas and Pitfalls
  - Solve Output by Type
  - Physics
     - The Physics Vector
     - Classical Mechanics
     - Introduction to Biomechanics Modeling
  - SymPy Special Topics
  - List of active deprecations
  - Glossary

I think it should match the module layout:

- Explanations
  - Best Practices
  - Gotchas and Pitfalls
  - Solve Output by Type
  - Physics
     - Vector
       - <Vector topic 1>
       - <Vector topic 2>
       - ...
     - Classical Mechanics
       - <mechanics topic 1>
       - <mechanics topic 2>
       - ...
     - Biomechanics
       - Introduction to Biomechanics Modeling
       - <biomechanics topic 2>
       - ...
  - SymPy Special Topics
  - List of active deprecations
  - Glossary

**Comment by moorepants** [comment]
Also, check on the doctest failures. Seems the optional dependencies are not setup correctly after the move.

**Comment by moorepants** [comment]
Please also add release notes.

**Comment by eh111eh** [comment]
> This looks good. I have the directory name commetn above and then looking at:
> 
> * Explanations
>   
>   * Best Practices
>   * Gotchas and Pitfalls
>   * Solve Output by Type
>   * Physics
>     
>     * The Physics Vector
>     * Classical Mechanics
>     * Introduction to Biomechanics Modeling
>   * SymPy Special Topics
>   * List of active deprecations
>   * Glossary
> 
> I think it should match the module layout:
> 
> * Explanations
>   
>   * Best Practices
>   * Gotchas and Pitfalls
>   * Solve Output by Type
>   * Physics
>     
>     * Vector
>       
>       * <Vector topic 1>
>       * <Vector topic 2>
>       * ...
>     * Classical Mechanics
>       
>       * <mechanics topic 1>
>       * <mechanics topic 2>
>       * ...
>     * Biomechanics
>       
>       * Introduction to Biomechanics Modeling
>       * <biomechanics topic 2>
>       * ...
>   * SymPy Special Topics
>   * List of active deprecations
>   * Glossary

I changed the directories so now the doc will look like:
* Physics
  * Vector
     * Vector & ReferenceFrame
     * ...
  * Classical Mechanics
     * Potential Issues/Advanced Topics/Future Features in Physics/Mechanics
     * ...
  * Biomechanics
    * Introduction to Biomechanical Modeling

**Comment by eh111eh** [comment]
> Also, check on the doctest failures. Seems the optional dependencies are not setup correctly after the move.

I keep failing the doctest as follows, but I'm uncertain about what is causing this error.
```
File "/home/runner/work/sympy/sympy/doc/src/explanation/Modules/Physics/mechanics/autolev_parser.rst", line 82, in autolev_parser.rst
Failed example:
    sympy_code = parse_autolev(open('double_pendulum.al'), include_numeric=True)
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/doctest.py", line 1368, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest autolev_parser.rst[1]>", line 1, in <module>
        sympy_code = parse_autolev(open('double_pendulum.al'), include_numeric=True)
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^
    FileNotFoundError: [Errno 2] No such file or directory: 'double_pendulum.al'
**********************************************************************
File "/home/runner/work/sympy/sympy/doc/src/explanation/Modules/Physics/mechanics/autolev_parser.rst", line 87, in autolev_parser.rst
Failed example:
    print(sympy_code)
Exception raised:
    Traceback (most recent call last):
      File "/opt/hostedtoolcache/Python/3.12.4/x64/lib/python3.12/doctest.py", line 1368, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest autolev_parser.rst[2]>", line 1, in <module>
        print(sympy_code)
              ^^^^^^^^^^
    NameError: name 'sympy_code' is not defined
**********************************************************************
1 items had failures:
   2 of   3 in autolev_parser.rst
***Test Failed*** 2 failures.
```

Also, I found that there are other errors in the doctest outside of the `autolev_parser.rst file`. However, these errors do not appear on my local laptop, which makes it a bit confusing to figure out.

**Comment by eh111eh** [comment]
Also the doc builds without any issues on my local laptop, so I'm also uncertain why it fails to build the doc.

Screenshot 2024-06-28 at 00 54 17

**Comment by moorepants** [comment]
Maybe this file:

```
    FileNotFoundError: [Errno 2] No such file or directory: 'double_pendulum.al'
```

was not moved with the other rst files.

**Comment by eh111eh** [comment]
> Maybe this file:
> 
> ```
>     FileNotFoundError: [Errno 2] No such file or directory: 'double_pendulum.al'
> ```
> 
> was not moved with the other rst files.

There's one `double_pendulum.al` file in `sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.al`, but there is no such an al file in the original location of `autolev_parser.rst`, `doc/src/modules/physics/mechanics/`. So I tried to create the `double_pendulum.al` file based on the example introduced in the `autolev_parser.rst` file and the one located in `sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.al` in the directory of `autolev_parser.rst`, but both al files don't seem to match with this rst file.
<br/>
al files I created in `doc/src/modules/physics/mechanics/` to fix this error:
- copy of sympy/parsing/autolev/test-examples/pydy-example-repo/double_pendulum.al
   ```
   MOTIONVARIABLES' Q{2}', U{2}'
   CONSTANTS L,M,G
   NEWTONIAN N
   FRAMES A,B
   SIMPROT(N, A, 3, Q1)
   SIMPROT(N, B, 3, Q2)
   W_A_N>=U1*N3>
   W_B_N>=U2*N3>
   POINT O
   PARTICLES P,R
   P_O_P> = L*A1>
   P_P_R> = L*B1>
   V_O_N> = 0>
   V2PTS(N, A, O, P)
   V2PTS(N, B, P, R)
   MASS P=M, R=M
   Q1' = U1
   Q2' = U2
   GRAVITY(G*N1>)
   ZERO = FR() + FRSTAR()
   KANE()
   INPUT M=1,G=9.81,L=1
   INPUT Q1=.1,Q2=.2,U1=0,U2=0
   INPUT TFINAL=10, INTEGSTP=.01
   CODE DYNAMICS() some_filename.c
  ```

- copy of double_pendulum.al example introduced in autolev_parser.rst file
   ```
   MOTIONVARIABLES' Q{2}', U{2}'
   CONSTANTS L,M,G
   NEWTONIAN N
   FRAMES A,B
   SIMPROT(N, A, 3, Q1)
   % -> N_A = [COS(Q1), -SIN(Q1), 0; SIN(Q1), COS(Q1), 0; 0, 0, 1]
   SIMPROT(N, B, 3, Q2)
   % -> N_B = [COS(Q2), -SIN(Q2), 0; SIN(Q2), COS(Q2), 0; 0, 0, 1]
   W_A_N>=U1*N3>
   % -> W_A_N> = U1*N3>
   W_B_N>=U2*N3>
   % -> W_B_N> = U2*N3>
   POINT O
   PARTICLES P,R
   P_O_P> = L*A1>
   % -> P_O_P> = L*A1>
   P_P_R> = L*B1>
   % -> P_P_R> = L*B1>
   V_O_N> = 0>
   % -> V_O_N> = 0>
   V2PTS(N, A, O, P)
   % -> V_P_N> = L*U1*A2>
   V2PTS(N, B, P, R)
   % -> V_R_N> = L*U1*A2> + L*U2*B2>
   MASS P=M, R=M
   Q1' = U1
   Q2' = U2
   GRAVITY(G*N1>)
   % -> FORCE_P> = G*M*N1>
   % -> FORCE_R> = G*M*N1>
   ZERO = FR() + FRSTAR()
   % -> ZERO[1] = -L*M*(2*G*SIN(Q1)+L*(U2^2*SIN(Q1-Q2)+2*U1'+COS(Q1-Q2)*U2'))
   % -> ZERO[2] = -L*M*(G*SIN(Q2)-L*(U1^2*SIN(Q1-Q2)-U2'-COS(Q1-Q2)*U1'))
   KANE()
   INPUT M=1,G=9.81,L=1
   INPUT Q1=.1,Q2=.2,U1=0,U2=0
   INPUT TFINAL=10, INTEGSTP=.01
   CODE DYNAMICS() some_filename.c
   ```

**Comment by moorepants** [comment]
There are a couple of figures more from the api directory, e.g.:

...odules/physics/mechanics/api/PinJoint.svg → ...on/modules/physics/mechanics/PinJoint.svg


Are those also used in the api docs? If so, do they still display properly in the api docs?

**Comment by moorepants** [comment]
This seems good to me other than the question about the api doc images.

**Comment by moorepants** [comment]
Actually, there are two pages still in the api section that should be moved:
image

**Comment by moorepants** [comment]
If you click "The Physics Vector" in the API docs:
image

You get a 404.

**Comment by moorepants** [comment]
@eh111eh When you build the docs locally, do a `make clean` and then `make html` then make sure to manually go through and check every page and every section link to make sure things are working.

**Comment by moorepants** [comment]
I think the headings need adjustment in the API sections:
image


We basically only need:

* API
  * Physics
    * Vector
    * Mechanics
    * Biomechanics

**Comment by eh111eh** [comment]
> There are a couple of figures more from the api directory, e.g.:
> 
> ...odules/physics/mechanics/api/PinJoint.svg → ...on/modules/physics/mechanics/PinJoint.svg
> 
> Are those also used in the api docs? If so, do they still display properly in the api docs?

`PinJoint.svg` file is used in `joints.rst` which is currently in `doc/src/explanation/modules/physics/mechanics/joints.rst`) and I don't think this svg file is used in api directory anymore.

**Comment by eh111eh** [comment]
> I think the headings need adjustment in the API sections: image
> 
> We basically only need:
> 
> * API
>   
>   * Physics
>     
>     * Vector
>     * Mechanics
>     * Biomechanics

Currently we have
```
API Reference
   - Physics
      - Hydrogen Wavefunctions
      - Matrices
      - Pauli Algebra
      - Quantum Harmonic Oscillator in 1-D
      - Quantum Harmonic Oscillator in 3-D
      - Second Quantization
      - Wigner Symbols
      - Unit systems
      - High energy physics
      - The Physics Vector Module
      - Classical Mechanics
      - Biomechanics
      - Quantum Mechanics
      - Optics Modules
      - Control Module
      - Continuum Mechanics
```
Considering the structure we need for api reference/physics, how about adjusting like either
```
API Reference
   - Physics
      - Vector
      - Mechanics
         - Quantum Mechanics
         - Continuum Mechanics
         - ...
      - Biomechanics
      - Special Topics
         - Hydrogen Wavefunctions
         - Matrices
         - Pauli Algebra
         - Quantum Harmonic Oscillator in 1-D
         - Quantum Harmonic Oscillator in 3-D
         - Second Quantization
         - Wigner Symbols
         - High energy physics
      - Unit systems
      - Optics
      - Control
```
or
```
API Reference
   - Physics
      - Vector
      - Mechanics
         - Quantum Mechanics
         - Continuum Mechanics
         - ...
      - Biomechanics
      - Special Topics
         - Hydrogen Wavefunctions
         - Matrices
         - Pauli Algebra
         - Quantum Harmonic Oscillator in 1-D
         - Quantum Harmonic Oscillator in 3-D
         - Second Quantization
         - Wigner Symbols
         - High energy physics
         - Unit systems
         - Optics
         - Control
```

**Comment by eh111eh** [comment]
I rearranged the files in API Reference/Physics so now the directory is
```
API Reference
   - Physics
      - Vector
      - Mechanics
         - Quantum Mechanics
         - Continuum Mechanics
         - ...
      - Biomechanics
      - Special Topics
         - Hydrogen Wavefunctions
         - Matrices
         - Pauli Algebra
         - Quantum Harmonic Oscillator in 1-D
         - Quantum Harmonic Oscillator in 3-D
         - Second Quantization
         - Wigner Symbols
         - High energy physics
      - Control
      - Optics
      - Unit Systems
```

I also checked that all pages display well with no error/problem
Screenshot 2024-07-04 at 18 20 12

**Comment by tjstienstra** [comment]
> > There are a couple of figures more from the api directory, e.g.:
> > ...odules/physics/mechanics/api/PinJoint.svg → ...on/modules/physics/mechanics/PinJoint.svg
> > Are those also used in the api docs? If so, do they still display properly in the api docs?
> 
> `PinJoint.svg` file is used in `joints.rst` which is currently in `doc/src/explanation/modules/physics/mechanics/joints.rst`) and I don't think this svg file is used in api directory anymore.

It is being used [here](https://github.com/sympy/sympy/blob/c26a4ead225820ebae27df3f67c81c78f70a463d/sympy/physics/mechanics/joint.py#L560).

**Comment by tjstienstra** [comment]
> I rearranged the files in API Reference/Physics so now the directory is

I would advise against making a special topics directory in physics. It is highly subjective and it is better to follow the module structure.

**Comment by eh111eh** [comment]
> > > There are a couple of figures more from the api directory, e.g.:
> > > ...odules/physics/mechanics/api/PinJoint.svg → ...on/modules/physics/mechanics/PinJoint.svg
> > > Are those also used in the api docs? If so, do they still display properly in the api docs?
> > 
> > 
> > `PinJoint.svg` file is used in `joints.rst` which is currently in `doc/src/explanation/modules/physics/mechanics/joints.rst`) and I don't think this svg file is used in api directory anymore.
> 
> [Here](https://github.com/sympy/sympy/blob/c26a4ead225820ebae27df3f67c81c78f70a463d/sympy/physics/mechanics/joint.py#L560) it is used.

I just updated `joint.py` file with new `PinJoint.svg` image file directory

**Comment by eh111eh** [comment]
> > I rearranged the files in API Reference/Physics so now the directory is
> 
> I would advise against making a special topics directory in physics. It is highly subjective and it is better to follow the module structure.

I tried to organise the files to follow the structure of:
API
- Physics
   - Vector
   - Mechanics
   - Biomechanics

But then, aside from these files, what would be the best way to organise the control, optics, units files and files currently in special topics category?

**Comment by moorepants** [comment]
> I would advise against making a special topics directory in physics. It is highly subjective and it is better to follow the module structure.

I agree with this. We should just follow the module structure (where a module is a folder in the source code with an `__init__.py`).

So the API doc: are:

- physics
  - module1
  - module2
  - module3
  - ...

Each module are directories in: https://github.com/sympy/sympy/tree/master/sympy/physics

**Comment by eh111eh** [comment]
I believe now api doc follows the module structure

**Comment by moorepants** [comment]
Couple more comments.

In the API docs we see:
image

It seems the header "Mechanics" and "Mechanics API Reference" are redundant. They can be merged into one index.rst for the mechanics API. Same for Vector API.

Secondly, do we have control of the order of these:

image


The "issues" pages should be at the end of the llsts and the two autolev pages should be grouped. I think we had a reasonable order for these already. Can that be maintained?

**Comment by eh111eh** [comment]
> Couple more comments.
> 
> In the API docs we see: image
> 
> It seems the header "Mechanics" and "Mechanics API Reference" are redundant. They can be merged into one index.rst for the mechanics API. Same for Vector API.
> 
> Secondly, do we have control of the order of these:
> 
> image
> 
> The "issues" pages should be at the end of the llsts and the two autolev pages should be grouped. I think we had a reasonable order for these already. Can that be maintained?

Thank you for pointing them out. I removed the additional `index.rst` files:
Screenshot 2024-07-06 at 23 37 33
Screenshot 2024-07-06 at 23 37 45

I also rearranged the order of `explanation/vector` and `explanation/classical mechanics` to the previous order. The only change from the original order is that the potential issues pages are now at the end:
Screenshot 2024-07-06 at 23 42 28
Screenshot 2024-07-06 at 23 42 42

**Comment by moorepants** [comment]
This looks good to me. Nice work! Lots of files to change and keep track of here.

**Pull request merged**